syntax = "proto3";
package robotics_msgs;
//option optimize_for = LITE_RUNTIME;

message Axis
{
    int32 vertical = 1;
    int32 horizontal = 2;
}

message PS3Axis {
    uint64 time = 1;
    uint32 l2_axis = 2;
    uint32 r2_axis = 3;
    Axis left_axis = 4;
    Axis right_axis = 5;
}

message PS3Button
{
    uint64 time = 1;
    bool l1_pressed = 2;
    bool r1_pressed = 3;
    bool l2_pressed = 4;
    bool r2_pressed = 5;
    bool left_pressed = 6;
    bool right_pressed = 7;
    bool up_pressed = 8;
    bool down_pressed = 9;
    bool select_pressed = 10;
    bool start_pressed = 11;
    bool ps_pressed = 12;
    bool l3_pressed = 13;
    bool r3_pressed = 14;
    bool triangle_pressed = 15;
    bool circle_pressed = 16;
    bool cross_pressed = 17;
    bool square_pressed = 18;
}

